domain/config/config_service.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"slices"
+	"sync"
+
+	backendconfig "github.com/bitcoin-sv/spv-wallet-web-backend/config"
+	"github.com/bitcoin-sv/spv-wallet-web-backend/domain/users"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/rs/zerolog"
+	"github.com/spf13/viper"
+)
+
+// ConfigService is a service for fetching and caching SharedConfig from the spv-wallet and providing PublicConfig.
+type ConfigService struct {
+	adminWalletClient users.AdminWalletClient
+	log               *zerolog.Logger
+
+	sharedConfig *models.SharedConfig
+	publicConfig *PublicConfig
+	mutex        sync.Mutex
+}
+
+// NewConfigService creates a new ConfigService.
+func NewConfigService(adminWalletClient users.AdminWalletClient, log *zerolog.Logger) *ConfigService {
+	return &ConfigService{
+		adminWalletClient: adminWalletClient,
+		log:               log,
+		sharedConfig:      nil,
+		publicConfig:      nil,
+	}
+}
+
+// GetSharedConfig returns shared config.
+// If shared config is not cached, it will be fetched from the spv-wallet.
+// SharedConfig should not be exposed to the public - use PublicConfig instead.
+func (s *ConfigService) GetSharedConfig() *models.SharedConfig {
+	if s.sharedConfig != nil {
+		return s.sharedConfig
+	}
```

Comment:
Actually the shared config can change in a meantime because it is taken from spv-wallet.
For the sake of simplicity I would rather prefer to not cache it and not store it here.

Reply:
I applied cache with TTL as we discussed.